pkg/apis/pipeline/v1alpha1/condition_types.go

```diff
@@ -73,6 +73,26 @@ type ConditionSpec struct {
 	// is evaluated
 	// +optional
 	Params []ParamSpec `json:"params,omitempty"`
+
+	// Resources is a list of the ConditionResources required to run the condition.
+	// +optional
+	Resources []ConditionResource `json:"resources,omitempty"`
+}
+
+// ConditionResource defines an input PipelineResource declared as a requirement
+// by a Task. The Name field will be used to refer to these Resources within
+// the Condition definition and  will be the path to the volume mounted containing this
+// Resource as an input (e.g. an Resource named `workspace` will be mounted at `/workspace`).
+type ConditionResource struct {
+	// Name declares the name by which a resource is referenced in the Condition's
+	// definition.
+	Name string `json:"name"`
+	// Type is the type of this resource;
+	Type PipelineResourceType `json:"type"`
+	// TargetPath is the path in workspace directory where the resource
+	// will be copied.
+	// +optional
+	TargetPath string `json:"targetPath"`
```

Comment:
when i first saw this i wondered, why not use the existing type?

https://github.com/tektoncd/pipeline/blob/bb855711f077b43c7948f37f5406bff69b4b29de/pkg/apis/pipeline/v1alpha1/task_types.go#L109-L122

I'm guessing it's b/c of the `OutputImageDir` field on the condition resource, is that correct? if that's the only reason then im thinking maybe we could do something like:

- Define one type (maybe called something like `ResourceInput`? or something better XD)
- Use this new type directly for conditions
- Make it so that `TaskResource` embeds the type and adds `OutputImageDir`
- Maybe open an issue to explore changing `OutputImageDir` (I'm not a big fan of the fact that the generic struct `TypeResource` has an image specific field :S)

Reply:
> Maybe open an issue to explore changing OutputImageDir (I'm not a big fan of the fact that the generic struct TypeResource has an image specific field :S)

We can probably leave that for now - between [the resource extensibility design](https://docs.google.com/document/d/1rcMG1cIFhhixMSmrT734MBxvH3Ghre9-4S2wbzodPiU) and [a proposal im making for treating output images differently](https://docs.google.com/document/d/1fF2vWMs12d3FwkqkNuzS7FzQFQSFz1ZKUe6CHqkIg0c/edit#heading=h.waoz4clirr5h) i think we're gonna get it fixed :D

but in the meantime embedding might be an okay solution!